src/utils/FixedQueue.ts

```diff
@@ -0,0 +1,55 @@
+import * as _ from 'underscore';
```

Comment:
would definitely delete the methods  `contains`, `getFirstElement`, and `removeElement`

contains --> call get queue and use `_.contains`
getFirstElement --> call getQueue()[0]  (you already do this in your code)
removeElement --> error prone, won't work with objects or arrays, plus you don't use it in your code

your idea of this datastructure is very nice though 👍